sdk/core/core-http/src/policies/disableResponseDecompressionPolicy.browser.ts

```diff
@@ -16,7 +16,9 @@ const DisbleResponseDecompressionNotSupportedInBrowser = new Error(
  * {@link DisableResponseDecompressionPolicy} is not supported in browser and attempting
  * to use it will results in error being thrown.
  */
-export function disableResponseDecompressionPolicy() {
+export function disableResponseDecompressionPolicy(): {
```

Comment:
can you use `RequestPolicyFactory` for the return type? same for the other factories below.

Reply:
Ooh thank you! I wasn't aware of this.

Reply:
@jeremymeng I couldn't do this without asking first:

This changes the public API, which can be seen in the `review/core-http.api.md` . The diff would have the following:

```diff
The file will have its original line endings in your working directory.
diff --git a/sdk/core/core-http/review/core-http.api.md b/sdk/core/core-http/review/core-http.api.md
index ee9253675..9760d303f 100644
--- a/sdk/core/core-http/review/core-http.api.md
+++ b/sdk/core/core-http/review/core-http.api.md
@@ -220,9 +220,7 @@ export class DisableResponseDecompressionPolicy extends BaseRequestPolicy {
 }
 
 // @public
-export function disableResponseDecompressionPolicy(): {
-    create: (nextPolicy: RequestPolicy, options: RequestPolicyOptions) => DisableResponseDecompressionPolicy;
-};
+export function disableResponseDecompressionPolicy(): RequestPolicyFactory;
 
 // @public
 export function encodeUri(uri: string): string;
@@ -382,9 +380,7 @@ export class KeepAlivePolicy extends BaseRequestPolicy {
 }
 
 // @public (undocumented)
-export function keepAlivePolicy(keepAliveOptions?: KeepAliveOptions): {
-    create: (nextPolicy: RequestPolicy, options: RequestPolicyOptions) => KeepAlivePolicy;
-};ore--
+export function keepAlivePolicy(keepAliveOptions?: KeepAliveOptions): RequestPolicyFactory;
 
 // @public (undocumented)
 export function logPolicy(loggingOptions?: LogPolicyOptions): RequestPolicyFactory;
```

If that's ok, I can push this change.

Reply:
`RequestPolicyFactory` is defined to be:

```
{
  create(nextPolicy: RequestPolicy, options: RequestPolicyOptionsLike) => RequestPolicy
}
```

Unfortunately this would be breaking. Widening the parameter from  `RequestPolicyOptions` -> `RequestPolicyOptionsLike` is fine, but widening the output type isn't from an API compatibility standpoint.

Reply:
Technically yes it's breaking. However I don't think it would have real impact though. I believe `create()` is only used in service client to create the request policy factories.  @xirzec any comments?

Also it's fine to undo this particular change to unblock PR.

Reply:
In the future version, we basically do this (policies return a generic type.) 

I don't think this will in practice break anyone as long as the consumers of the policies accept `RequestPolicy`, which I believe they all do. I don't think we take `KeepAlivePolicy` as an input type anywhere, for example.

Reply:
Lets make the change to return `RequestPolicyFactory`